src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -78,9 +74,10 @@ public function onKernelResponse(ResponseEvent $event)
         // Always remove the internal header if present
         $response->headers->remove(self::NO_AUTO_CACHE_CONTROL_HEADER);
 
-        if (!$session = $this->container && $this->container->has('initialized_session') ? $this->container->get('initialized_session') : $event->getRequest()->getSession()) {
```

Comment:
Can you shed my light on why we don't need `initialized_session` anymore?
IIRC, it was there to skip accessing the session when it's not been used before.

Reply:
Added a local property to track usage of the Factory. That prevent calling `$request->getSession()` for nothing in the `onKernelResponse` method.

Beware. This also breaks the behavior if the user sets its own session in the request (which is already the case in symfony < 6 that use `initialized_session`).

- onKernelRequest => the factory is injected in the request
- user call `request->setSession(new Session())`
- onKernelResponse realize that the factory has not been called => early exit

Reply:
WDYT of `hasSession(bool $skipIfUninitialized = false)`, with a BC layer on 5.4?

Reply:
See #42900